img_proof/ipa_cloud.py

```diff
@@ -369,7 +369,11 @@ def _run_test(self, test, ssh_config):
 
         while num_retries < self.retry_count:
             plugin = Report()
-            result = pytest.main(cmds, plugins=[plugin])
+
+            try:
+                result = pytest.main(cmds, plugins=[plugin])
+            except Exception:
+                result = 2
```

Comment:
Why 2?

Reply:
Pytest returns 1 if tests fail, 0 if sucess and any of 2, 3, 4 if there's some sort of handled exception. So 2 is arbitrary, it could be 2, 3 or 4. Below this section where the results are handled if the result is in (2, 3, 4) img-proof only adds a generic error instead of trying to parse the plugin report which won't exist in exception cases.

Reply:
Consider adding that to the README and/or documentation.

Reply:
Will add comments before merge.